core-bundle/src/Security/DataContainer/AbstractAction.php

```diff
@@ -44,7 +44,7 @@ public function __construct(private string $dataSource)
 
     public function __toString(): string
     {
-        return sprintf('[Subject: %s]', implode('; ', $this->getSubjectInfo()));
+        return sprintf('[Subject: %s; %s]', ltrim(strrchr(get_called_class(), '\\'), '\\'), implode('; ', $this->getSubjectInfo()));
```

Comment:
This greatly improves debug infos if a DCA permission fails. It currently does not tell what action failed (create, read, update, delete), which would be really helpful imho 😆